### ISSUE TITLE
[meta] Tighten up and enforce stakeholder requirements

### ISSUE THREAD

**Issue by DanielOaks** [ircv3/ircv3-specifications#222]
So, #215 gives stakeholders more well-defined powers in the IRCv3 WG. Namely, the ability to put forth specs for discussion and inclusion into IRCv3.

If we go through with the process changes set out by #215, we need to have a proper way of accepting stakeholders to the IRCv3 org, probably just through issues on this repo. The primary 'list' of stakeholders will probably be contained in the website repo as a YAML document, so we can display a list of stakeholders on there, so we would either discuss it directly on PRs to the website repo or as an issue here first, then the people that handle the site would commit changes as necessary.

Should there be some sort of a voting system when designating stakeholders, along with the requirements? Should stakeholders also be able to include some users that don't necessarily have the widespread implementation requirements set out by the charter right now, but have shown goodwill and participation to the WG?

Basically, we need a document to set this out concretely.

**edit:** They should also be given access to the IRCv3 Github org as a 'stakeholder' (ie, probably no permissions at all) to show their status on issue trackers and such.

**Comment by Aerdan** [comment]
Nonwithstanding other requirements, individuals who have already contributed materially to the IRCv3 organization should retain stakeholdership, unless they are removed for cause.

**Comment by DanielOaks** [comment]
I think that's reasonable.

Given the 'sponsoring' that #215's process allows, if someone gets a spec merged into the IRCv3 repo (for instance, one that was sponsored by an IRCv3 stakeholder), I think them gaining stakeholder status (or at least being considered for it) wouldn't be that bad.

**Comment by attilamolnar** [comment]
> Nonwithstanding other requirements, individuals who have already contributed materially to the IRCv3 organization should retain stakeholdership, unless they are removed for cause.

> ...if someone gets a spec merged into the IRCv3 repo (for instance, one that was sponsored by an IRCv3 stakeholder), I think them gaining stakeholder status (or at least being considered for it / considered to have fulfilled the 'implementation' requirements of becoming a stakeholder) wouldn't be that bad.

I'm ok with letting the mentioned people (already contributed in the past, will contribute a spec in the future) open issues on the main repo without requiring a sponsor, however allowing the mentioned people to vote on current issues is one step too far, I believe. Only people close to the fire, people running large networks and vendors who develop notable IRC software should be considered when voting is necessary to resolve an issue.

A new category could be added for this.

**Comment by DarthGandalf** [comment]
There is a fine line which software is notable enough and which networks
are large enough...

2016-02-12 14:23 GMT+00:00 Attila Molnar <notifications@github.com>:

**Comment by jwheare** [comment]
However, there are some networks that are very obviously small and software that's very obviously not notable. The fine line can be debated when it occurs, but broad strokes are still be useful.

**Comment by highflyer77** [comment]
As in other organizations if a current stockholder nominates someone to be a member (and would probably state the reasons why), and another person seconds it. Then a vote would be taken and the person would be accepted or denied membership.

**Comment by DanielOaks** [comment]
@attilamolnar Fair. Honestly, the main issue is just making sure that we have a reasonable process for accepting new stakeholders. Working out the list of *current* stakeholders is something odd that will basically only happen once, so we can decide the requirements and such for that basically however we want.
